### PR TITLE
feat: add synapse-invite-checker

### PR DESCRIPTION
**Pull request by amandahla**
### Overview

This PR adds the synapse-invite-checker module.

https://git.buechner.me/nbuechner/synapse-invite-checker/src/branch/main

### Rationale

Moderate invites.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Manging Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [ ] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Review by yanksyoon — APPROVED** [review]
LGTM!

**Review by javierdelapuente — APPROVED** [review]

File: src/charm_state.py [hunk 1]
```diff
@@ -246,6 +250,31 @@ def to_yes_or_no(cls, value: str) -> str:
...
+
+        Returns:
+            The string converted to list.
+
+        Raises:
+            ValidationError: if rooms is not as expected.
+        """
+        # Based on documentation
+        # https://spec.matrix.org/v1.10/appendices/#user-identifiers
+        roomid_regex = r"![a-z0-9._=/+-]+:\w+\.\w+"
```
[the remark this answers is not in the document]
Comment (amandahla) [reply src/charm_state.py:269]: The room ID should be like "!opaque_id:domain"
"!fTjMjIzNKEsFlUIiru:server123.com"

The config is expected to be "fTjMjIzNKEsFlUIiru:server123.com

(I forgot to change the link in the comment)
https://spec.matrix.org/v1.10/appendices/#room-ids

[the remark this answers is not in the document]
Comment (amandahla) [reply src/charm_state.py:269]: I can change the regex to expected more than 2 parts like "name1.name2.com". Is that what you mean?

[the remark this answers is not in the document]
Comment (javierdelapuente) [reply src/charm_state.py:269]: Yeah, I mean that I think a server domain could have mote parts like test.ubuntu.com

[the remark this answers is not in the document]
Comment (amandahla) [reply src/charm_state.py]: good catch, done.

**Review by javierdelapuente — APPROVED** [review]

**Review by yanksyoon — APPROVED** [review]

**Pull request merged**